ballerina/natives.bal

```diff
@@ -123,6 +136,31 @@ public isolated function printWarn(string msg, error? 'error = (), *KeyValues ke
     }
 }
 
+# Set the log output to a file. Note that all the subsequent logs of the entire application will be written to this file.
+# ```ballerina
+# log:setOutputFile("./resources/myfile.log");
+# log:setOutputFile("./resources/myfile.log", log:APPEND);
+# ```
+#
+# + path - The path of the file
+# + option - To indicate whether to overwrite or append the log output
+public isolated function setOutputFile(string path, FileWriteOption option = OVERWRITE) {
+    lock {
+        outputFilePath = path;
+    }
+    lock {
+        fileWriteOutputOption = option;
+    }
+    lock {
+        if option == OVERWRITE {
+            io:Error? result = io:fileWriteString(path, "");
+            if result is error {
+                printError("failed to set log output file", 'error = result);
+            }
+        }
+    }
```

Comment:
Do we need this `lock` statement block ?

Reply:
Removed in 98b4983